Source/Diagnostics/FlushFormats/FlushFormatCatalyst.H

```diff
@@ -0,0 +1,56 @@
+#ifndef WARPX_FLUSHFORMATCATALYST_H_
+#define WARPX_FLUSHFORMATCATALYST_H_
+
+#include "FlushFormatInSitu.H"
+
+#include "Diagnostics/ParticleDiag/ParticleDiag_fwd.H"
+
+#ifdef AMREX_USE_CONDUIT
+#   include <AMReX_Conduit_Blueprint.H>
+#endif
+#include <AMReX_Geometry.H>
+#include <AMReX_Vector.H>
+
+#include <AMReX_BaseFwd.H>
+
+#ifdef AMREX_USE_CATALYST
+#   include <catalyst.hpp>
+#endif
+
+#include <string>
+
+/**
+ * \brief This class aims at dumping performing in-situ analysis and visualization
+ * with Catalyst. Catalyst initialize and finalize are called in the constructor
+ * and destructor respectivelyThe function WriteToFile takes in the particles,
+ * meshes, and fields, and formats them to be used by given pipeline scripts. All
+ * exports are defined and executed externally through the given catalyst pipeline
+ * and implementation.
+ */
+class FlushFormatCatalyst : public FlushFormatInSitu
+{
+public:
+    // Initialize
+    FlushFormatCatalyst();
```

Comment:
Clang-Tidy spotted:
```
/home/runner/work/WarpX/WarpX/Source/Diagnostics/FlushFormats/FlushFormatCatalyst.H:30:7: error: class 'FlushFormatCatalyst' defines a non-default destructor but does not define a copy constructor, a copy assignment operator, a move constructor or a move assignment operator [cppcoreguidelines-special-member-functions,-warnings-as-errors]
class FlushFormatCatalyst : public FlushFormatInSitu
      ^
```

Reply:
Let's add default or deleted copy/move constructors (see `FlushFormatAscent`) to obey the rule-of-five?